edb/graphql/translator.py

```diff
@@ -373,10 +377,7 @@ def _should_include(self, directives):
 
                 if isinstance(cond, gql_ast.VariableNode):
                     varname = cond.name.value
-                    var = self._context.vars[varname]
-                    # mark the variable as critical
-                    self._context.vars[varname] = var._replace(critical=True)
-                    value = var.val
+                    value = self._context.vars[varname].val
```

Comment:
"Critical" variables used to be part of the cache key (because they affect query shape). How is caching of different shapes due to include/skip directive variables handled now?

I tried to run queries in GraphiQL in this branch and it appears that there's no problem in correctly handling these variables, but I wonder what the mechanism is. The reason is that if this mechanism uses something other than this "critical" flag, we should probably drop the code handling it.

Reply:
Unfortunately, they weren't part of a cache key. There was (and still is) a merely if statement that always recompiled if there is a critical variables.

Currently variables from the include/exclude are present in the cache key:

Reply:
Old mechanism, which is still here:
https://github.com/edgedb/edgedb/blob/16f43cd44383373908f06eedca92923fbbb74d72/edb/server/http_graphql_port/protocol.pyx#L236
Is still used for introspection queries. I would like this to be handled in some other way, but we can do it in a separate PR.